File: 2611-mice-and-cheese/2611-mice-and-cheese.java
class Solution {
    public int miceAndCheese(int[] reward1, int[] reward2, int k) {
        int n= reward1.length;
        
        Queue<Integer> maxHeap = new PriorityQueue<Integer>(Comparator.reverseOrder());
        for(int i=0; i<n; i++) {
            maxHeap.offer(reward1[i] - reward2[i]);
        }
        
        int totalScore = 0;
        for(int i=0; i<n; i++) {
            totalScore += reward2[i];
        }
        
        while(k > 0) {
            totalScore += maxHeap.poll();
            k--;
        }
        
        return totalScore;
    }
}

File: 2611-mice-and-cheese/2611-mice-and-cheese.cpp
class Solution {
public:
    int miceAndCheese(vector<int>& reward1, vector<int>& reward2, int k) {
        priority_queue<pair<int, int>> maxHeap;
        int n = reward1.size();
        
        int total = accumulate(reward2.begin(), reward2.end(), 0);
        
        for(int i=0; i<n; i++) {
            maxHeap.push(make_pair(reward1[i] - reward2[i], i));
        }
        
        while(k--) {
            total += maxHeap.top().first;
            maxHeap.pop();
        }
        
        return total;
    }
};